src/macros.jl

```diff
@@ -1736,102 +1736,99 @@ function _reorder_parameters(args)
-  `varexpr` constrained to belong to `set`, see [Variables constrained on creation](@ref).
-
-The expression `varexpr` can either be
-
-* of the form `varname` creating a scalar real variable of name `varname`;
-* of the form `varname[...]` or `[...]` creating a container of variables.
+positional arguments `args` and the keyword arguments `kw_args`.
+
+## Anonymous and named variables
+
+`expr` must be one of the forms:
+
+ * Omitted, like `@variable(model)`, which creates an anonymous variable
+ * A single symbol like `@variable(model, x)`
+ * A container expression like `@variable(model, x[i=1:3])`
+ * An anoymous container expression like `@variable(model, [i=1:3])`
+
+## Bounds
+
+In addition, the expression can have bounds, such as:
+
+ * `@variable(model, x >= 0)`
+ * `@variable(model, x <= 0)`
+ * `@variable(model, x == 0)`
+ * `@variable(model, 0 <= x <= 1)`
+
+and bounds can depend on the indices of the container expressions:
+
+ * `@variable(model, -i <= x[i=1:3] <= i)`
+
+## Sets
+
+You can explicitly specify the set to which the variable belongs:
+
+ * `@variable(model, x in MOI.Interval(0.0, 1.0))`
+
+ For more information on this syntax, read
+[Variables constrained on creation](@ref).
+
+## Positional arguments
 
 The recognized positional arguments in `args` are the following:
 
-* `Bin`: Sets the variable to be binary, i.e. either 0 or 1.
-* `Int`: Sets the variable to be integer, i.e. one of ..., -2, -1, 0, 1, 2, ...
-* `Symmetric`: Only available when creating a square matrix of variables, i.e.
-  when `varexpr` is of the form `varname[1:n,1:n]` or `varname[i=1:n,j=1:n]`.
-  It creates a symmetric matrix of variable, that is, it only creates a
-  new variable for `varname[i,j]` with `i ≤ j` and sets `varname[j,i]` to the
-  same variable as `varname[i,j]`. It is equivalent to using
-  `varexpr in SymmetricMatrixSpace()` as `expr`.
-* `PSD`: The square matrix of variable is both `Symmetric` and constrained to be
-  positive semidefinite. It is equivalent to using `varexpr in PSDCone()` as
-  `expr`.
+ * `Bin`: restricts the variable to the [`MOI.ZeroOne`](@ref) set, that is,
+   `{0, 1}`. For example, `@variable(model, x, Bin)`. Note: you cannot use
+   `@variable(model, Bin)`, use the `binary` keyword instead.
+ * `Int`: restricts the variable to the set of integers, that is, ..., -2, -1,
+    0, 1, 2, ... For example, `@variable(model, x, Int)`. Note: you cannot use
+    `@variable(model, Int)`, use the `integer` keyword instead.
+ * `Symmetric`: Only available when creating a square matrix of variables, i.e.,
+   when `expr` is of the form `varname[1:n,1:n]` or `varname[i=1:n,j=1:n]`,
+   it creates a symmetric matrix of variables.
+ * `PSD`: A restrictive extension to `Symmetric` which constraints a square
+   matrix of variables to `Symmetric` and constrains to be positive
+   semidefinite.
+
+## Keyword arguments
 
 The recognized keyword arguments in `kw_args` are the following:
```

Comment:
Maybe add that the kwargs are especially useful for the anon variables case.

Reply:
How's this now?

Reply:
great